src/components/Card.js

```diff
@@ -1,56 +1,179 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  ) {
     this._name = name;
     this._link = link;
+    this._id = _id;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this.handleDeleteClick = handleDeleteClick;
+    this.handleLikeClick = handleLikeClick;
+    this.isLiked = isLiked;
   }
 
   _setEventListeners() {
     //".card__like-button"
     this._cardElement
       .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeIcon();
+      .addEventListener("click", (e) => {
+        e.stopPropagation();
+        this.handleLikeClick(this);
       });
-    //".card__delete-button"
+    // (".card__delete-button");
     this._cardElement
       .querySelector(".card__delete-button")
-      .addEventListener("click", () => {
-        this._handleDeleteCard();
+      .addEventListener("click", (e) => {
+        e.stopPropagation();
+        // call the function passed as an argument
+        // pass this as argument
+        this.handleDeleteClick(this._id, this._cardElement);
       });
     // TODO - add listener for image element
-    this._cardImageEl.addEventListener("click", () => {
+    this._cardElement.addEventListener("click", (e) => {
+      e.stopPropagation();
       this._handleImageClick(this._name, this._link);
     });
   }
 
-  _handleLikeIcon() {
+  // make this a public method
+  // handleLikeIcon(isLiked) {
+  //   if (isLiked) {
+  //     this._cardElement
+  //       .querySelector(".card__like-button")
+  //       .classList.add("card__like-button_active");
+  //   } else {
+  //     this._cardElement
+  //       .querySelector(".card__like-button")
+  //       .classList.remove("card__like-button_active");
+  //   }
+  // }
+
+  // Use after successful like response (in a .then)
+  likeCard() {
+    this._cardElement
+      .querySelector(".card__like-button")
```

Comment:
**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form`, `this._likeButton`, and so on.

```js
  getView() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```